csrc/host_ir/lower.cpp

```diff
@@ -75,8 +75,8 @@ std::unique_ptr<hir::HostIrContainer> HostIrLower::lower(
     DeviceIdxType my_device_index) {
   // Sharding PreSegmenter passes.
   // Note: passes run before PreSegmenter optimization passes.
-  preseg_passes::OptimizationPass<
-      preseg_passes::PropagateShardingsPass>::runPass(fusion.get());
```

Comment:
Removing this to allow fusion inputs to have `Stream` parallel type.
I initially tried to make `StreamParallelType` pass be compatible with this, however, there are quite a few differences that will need to be resolved. It also does not allow for logical domain to be split to produce a `Stream` axis and requires it be in the logical domain.

Since `MultideviceExecutor` does not need this pass much (I only had to fix one test), it seemed simplest to not use it here. In the future, if we want this pass for `MultiDeviceExecutor`, we can: 
1. Have diverging `PropagateShardingsPass`
2. Make `MultiDeviceExecutor` compatible